polkadot/node/core/backing/src/lib.rs

```diff
@@ -1255,49 +1147,36 @@ async fn seconding_sanity_check<Context>(
 	let candidate_relay_parent = hypothetical_candidate.relay_parent();
 	let candidate_hash = hypothetical_candidate.candidate_hash();
 
-	for (head, leaf_state) in active_leaves {
-		if ProspectiveParachainsMode::from(leaf_state).is_enabled() {
-			// Check that the candidate relay parent is allowed for para, skip the
-			// leaf otherwise.
-			let allowed_parents_for_para =
-				implicit_view.known_allowed_relay_parents_under(head, Some(candidate_para));
-			if !allowed_parents_for_para.unwrap_or_default().contains(&candidate_relay_parent) {
-				continue
-			}
-
-			let (tx, rx) = oneshot::channel();
-			ctx.send_message(ProspectiveParachainsMessage::GetHypotheticalMembership(
-				HypotheticalMembershipRequest {
-					candidates: vec![hypothetical_candidate.clone()],
-					fragment_chain_relay_parent: Some(*head),
-				},
-				tx,
-			))
-			.await;
-			let response = rx.map_ok(move |candidate_memberships| {
-				let is_member_or_potential = candidate_memberships
-					.into_iter()
-					.find_map(|(candidate, leaves)| {
-						(candidate.candidate_hash() == candidate_hash).then_some(leaves)
-					})
-					.and_then(|leaves| leaves.into_iter().find(|leaf| leaf == head))
-					.is_some();
-
-				(is_member_or_potential, head)
-			});
-			responses.push_back(response.boxed());
-		} else {
-			if *head == candidate_relay_parent {
-				if let ActiveLeafState::ProspectiveParachainsDisabled { seconded } = leaf_state {
-					if seconded.contains(&candidate_para) {
-						// The leaf is already occupied. For non-prospective parachains, we only
-						// second one candidate.
-						return SecondingAllowed::No
-					}
-				}
-				responses.push_back(futures::future::ok((true, head)).boxed());
-			}
+	for head in implicit_view.all_allowed_relay_parents() {
```

Comment:
we only need to do this for active leaves. We only care if we can back something on an active leaf. Prospective parachains will take the possible ancestry into account

Reply:
Yes, I should use `leaves()` instead. Fixed.